fix bug where save size could be detected incorrectly, if multiple bytes in the save are identical

Peter0x44 opened the pull request:
Recently, a user of GodMode9i reported that it was always dumping only the first 8192 bytes of save, as the function cardEepromGetSize was reporting the wrong value, instead of the 64k which the save should have been. 

Flashing this save to another cartridge also caused the size to be detected incorrectly, so it was nothing with the flash type itself

What causes this here is that both the first and 8192nd words are 0, so it just stops looping early
By writing TeST (a very unlikely value to appear later on) and restoring it after finishing the loop, the size is now detected correctly.



[SCRIBLENAUT2_BH2EWR_0.zip](https://github.com/devkitPro/libnds/files/5872255/SCRIBLENAUT2_BH2EWR_0.zip)

urmum-69 commented:
I force pushed to change some formatting - I accidentally used spaces instead of tabs in some places

Commit:
Fix bug where the save size could be detected incorrectly, if multiple bytes in the save are identical

Commit:
Properly handle 128k size saves

Before, 128k saves (like those present in Pokemon Mystery Dungeon: Explorers of Sky would be detected incorrectly as 8k saves.
Also, the first byte read from them would always be 0xFF

Peter0x44 commented:
Another user ran in to a different issue with dumping saves - the code didn't handle 128k eeprom properly
It would get detected as 8k, and the dump would contain a single erroneous 0xFF byte at that start.

I'm going to throw that change in here, hopefully someone can get around to reviewing it soon.

Commit:
Use safer EEPROM size check, with no risk of false positive results
